CMSgov/design-system
# [WNMGDS-2807] Fix medicare component overrides and make handling of side effects consistent

## Summary

[WNMGDS-2807](https://jira.cms.gov/browse/WNMGDS-2807)

While we were trying to make component overrides and side-effects handling consistent between `ds-healthcare-gov` and `ds-medicare-gov`, we discovered that the three component overrides for medicare (`HelpDrawer`, `HelpDrawerToggle`, and `ThirdPartyExternalLink`) don't actually work, because they rely on side effects that aren't flagged for bundlers. In `ds-healthcare-gov`, we were marking the whole package as having side effects, but in `ds-medicare-gov`, we were only marking the web components as having side effects. This resulted in lost opportunity for tree-shaking on the healthcare side and a bug on the medicare side. Both packages have been updated to explicitly list the component-override modules (among others) as having side effects.

## How to test

### Testing `ds-healthcare-gov`
1. Check out 8d70996240a90b24bf79a0a1977e7f8ddeb351a0
2. Run `yarn build:healthcare`
3. Go to the `examples/react-app` project and build and run it.
4. You should be able to see the healthcare-specific Accordion designs and an external link modal with a healthcare-specific "learn more" link. You should also see the text field's error message on the bottom. 

### Testing `ds-medicare-gov`
1. Check out bafba45c6e24436b4e79ae2dac12b7cf73b7ba05
2. Run `yarn build:medicare`
3. Go to the `examples/react-app` project and build and run it.
4. You should be able to see the medicare-specific HelpDrawer designs in the toggle and the drawer itself and an external link modal with a medicare-specific "learn more" link. The text field doesn't matter, but I forgot to remove it.

## Checklist
